### ISSUE TITLE
False positive in "await-promise" rule for subtype of Promise (angular.IPromise)

### ISSUE THREAD

**Issue by Jack-Works** [palantir/tslint#3616]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI 

#### TypeScript code being linted

```ts
class YPromise<T> extends Promise<T> {}
async function name() {
    await new YPromise(() => {})
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "await-promise": { "severity": "warning" }
  }
}

```

#### Actual behavior
Invalid 'await' of a non-Promise value.

#### Expected behavior
Nothing.

### Seniaro
This happened in my angular app. After I making angular.IPromise a Promise subtype, tslint still warning about this.

```ts
declare module 'angular' {
    interface IPromise<T> extends Promise<T> {}
}
```

### Workaround
```json
"await-promise": { "severity": "warning", "options": ["IPromise"] }
```

**Comment by ajafff** [comment]
Duplicate of #3381, fixed by #3383, but not released yet
